cmd/skaffold/app/cmd/runner.go

```diff
@@ -128,13 +133,13 @@ func withFallbackConfig(out io.Writer, opts config.SkaffoldOptions, getCfgs func
 	return nil, fmt.Errorf("parsing skaffold config: %w", err)
 }
 
-func setDefaultDeployer(configs []*latestV1.SkaffoldConfig) {
+func setDefaultDeployer(configs parser.SkaffoldConfigSet) {
 	// do not set a default deployer in a multi-config application.
 	if len(configs) > 1 {
 		return
 	}
 	// there always exists at least one config
-	defaults.SetDefaultDeployer(configs[0])
+	defaults.SetDefaultDeployer(configs[0].SkaffoldConfig)
```

Comment:
nit: Is it possilbe that configs[0] is nil?  From the comment it doesn't seem possible, wanted to verify